modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/test/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImplTest.java

```diff
@@ -430,49 +444,88 @@ private void _testExtractIndexableAttributes(
 				LocaleUtil.US));
 	}
 
-	private void _testFormWithRepeatableField(String indexType) {
+	private void _testFormWithRepeatableField(
+		String indexType, boolean enableLegacyDDMIndexFields) {
+
 		Document document = _createDocument();
 
-		DDMForm ddmForm = DDMFormTestUtil.createDDMForm(
-			SetUtil.fromArray(LocaleUtil.US), LocaleUtil.US);
+		if (enableLegacyDDMIndexFields) {
+			_ddmIndexer = _createDDMIndexer(true);
 
-		DDMFormField ddmFormField = _createDDMFormField();
+			DDMForm ddmForm = DDMStructureTestUtil.getSampleDDMForm(
+				_FIELD_NAME, "string", "text", true,
+				DDMFormFieldTypeConstants.RICH_TEXT,
+				new Locale[] {LocaleUtil.getSiteDefault()},
+				LocaleUtil.getSiteDefault());
 
-		ddmFormField.setIndexType(indexType);
-		ddmFormField.setRepeatable(true);
-
-		ddmForm.addDDMFormField(ddmFormField);
+			DDMFormField ddmFormField = ddmForm.getDDMFormFields(
+			).get(
+				0
+			);
```

Comment:
@icgsbr you don't need to get the `DDMFormField` as you are only using this variable to get the its name. 

You can change the code to:

```
Assert.assertEquals(
	"test",
	document.get(
		StringBundler.concat(
			"ddm__text__", ddmStructure.getStructureId(), "__",
			_FIELD_NAME, "_en_US")));
```

since `_FIELD_NAME` is actually the `ddmFormField.getName()`